lib/Sema/MiscDiagnostics.cpp

```diff
@@ -1328,7 +1328,26 @@ void swift::fixItAvailableAttrRename(TypeChecker &TC,
     return labelStr.empty() ? Identifier() : TC.Context.getIdentifier(labelStr);
   });
 
-  if (auto args = dyn_cast<TupleExpr>(argExpr)) {
+  if (auto args = dyn_cast<TupleShuffleExpr>(argExpr)) {
```

Comment:
My implementation fails in the case:
```
@available(*, unavailable, renamed: "newFn(x:y:)")
func oldFunc(a: Int ..., b: Int = 0) { }
oldFunc(a: 1, 1)
^~~~~~~ ~
newFn   x     y: 

```
argExpr is
```
(tuple_shuffle_expr implicit type='(a: Int..., b: Int)' elements=[-2, -1] variadic_sources=[0, 1]
  (tuple_expr type='(a: Int, Int)' names=a,'' ...))
```
It should be more restrictive in TupleShuffleExpr case.